### PR TITLE
.github: enable asan/ubsan build and test

### PR DESCRIPTION
**Pull request by kylophone**
Adds an ASAN/UBSAN libvmaf build and test to the CI.

### CODE REVIEW

**Comment by kylophone** [comment]
- https://github.com/Netflix/vmaf/runs/4889311133 should fix the leaks in the tests.

**Comment by kylophone** [comment]
> https://github.com/Netflix/vmaf/runs/4889311133 should fix the leaks in the tests.

Pushed 511222075f9bbce332174f96b9f97dc3b493126b to master.

**Comment by kylophone** [comment]
```sh
src/liblibvmaf_cpu.a.p/cpu.c.o: In function `asan.module_ctor':
cpu.c:(.text+0x62): undefined reference to `__asan_init'
cpu.c:(.text+0x67): undefined reference to `__asan_version_mismatch_check_v8'
cpu.c:(.text+0x7a): undefined reference to `__asan_register_globals'
src/liblibvmaf_cpu.a.p/cpu.c.o: In function `asan.module_dtor':
cpu.c:(.text+0x90): undefined reference to `__asan_unregister_globals'
```

Almost there, just some linker errors in the `ubuntu-18.04 ccache clang` build.

File: .github/workflows/libvmaf.yml [hunk 1]
```diff
@@ -75,6 +75,15 @@ jobs:
       - name: Run tests
         run: |
           sudo ninja -vC libvmaf/build test
+      - name: Run meson asan
+        run: |
+          LDFLAGS=-fsanitize=address,undefined meson setup libvmaf libvmaf/build_asan --buildtype debug -Db_sanitize=address,undefined -Denable_float=true
+      - name: Run ninja asan
+        run: |
+          sudo ninja -vC libvmaf/build_asan install
+      - name: Run tests asan
```
Comment (nilfm) [inline .github/workflows/libvmaf.yml:84]: Are these tests a superset of "Run tests" above? as in, is there a way that `Run tests asan` could pass but `Run tests` could fail? I'm thinking that `Run tests` may be redundant

Comment (kylophone) [reply .github/workflows/libvmaf.yml:84]: The optimized binaries are available for download after the whole thing is done, that's why we do both. The latency/compute overhead should be small.

**Comment by nilfm** [comment]
I was able to reproduce the issue with this Dockerfile:

```
FROM ubuntu:18.04

WORKDIR /home

RUN cd /home
RUN apt update
RUN apt install -y git gcc g++ nasm xxd python3.7 python3-pip ninja-build clang
RUN pip3 install setuptools
RUN git clone https://github.com/mesonbuild/meson.git && \
    cd meson && \
    python3.7 setup.py install && \
    cd ..
RUN git clone https://github.com/Netflix/vmaf.git
RUN echo "CC=clang CXX=clang CFLAGS=-fsanitize=address,undefined CPPFLAGS=-fsanitize=address,undefined LDFLAGS=-fsanitize=address,undefined /usr/local/bin/meson build_asan --buildtype debug -Db_sanitize=address,undefined -Denable_float=true" > vmaf/libvmaf/run.sh && chmod +x vmaf/libvmaf/run.sh
CMD ["bash"]
```

Steps:
```
docker pull ubuntu:18.04
docker build -t myubuntu18 .
docker run -it myubuntu18
cd vmaf/libvmaf
./run.sh
ninja -vC build_asan
```

Mainly two errors:
```
/home/vmaf/libvmaf/build_asan/../src/feature/common/blur_array.c:206: undefined reference to `__ubsan_handle_pointer_overflow'
/home/vmaf/libvmaf/build_asan/../src/feature/common/blur_array.c:209: undefined reference to `__ubsan_handle_type_mismatch_v1'
```

I haven't been able to get it to work.

**Pull request closed without merging**